plugins/module_utils/foreman_helper.py

```diff
@@ -494,11 +494,8 @@ def find_resource(self, resource, search, params=None, failsafe=False, thin=None
             else:
                 error_msg = "no"
             self.fail_json(msg="Found {0} results while searching for {1} with {2}".format(error_msg, resource, search))
-        if result:
-            if thin:
-                result = {'id': result['id']}
-            else:
-                result = self.show_resource(resource, result['id'], params=params)
+        if result and not thin:
```

Comment:
You can't `show` an external auth source, so I can't pass `thin=False`, but at the same time, `index` already have us all the details and we should just not throw them away. `_flatten_entity` will take care of that when needed at a later point.